source/Octopus.Cli/Commands/Package/PushCommand.cs

```diff
@@ -22,18 +22,13 @@ public PushCommand(IOctopusAsyncRepositoryFactory repositoryFactory, IOctopusFil
             : base(clientFactory, repositoryFactory, fileSystem, commandOutputProvider)
         {
             var options = Options.For("Package pushing");
-            options.Add("package=", "Package file to push. Specify multiple packages by specifying this argument multiple times: \n--package package1 --package package2", 
+            options.Add<string>("package=", "Package file to push. Specify multiple packages by specifying this argument multiple times: \n--package package1 --package package2", 
                 package => Packages.Add(EnsurePackageExists(fileSystem, package)));
-            options.Add("overwrite-mode=", "If the package already exists in the repository, the default behavior is to reject the new package being pushed (FailIfExists). You can use the overwrite mode to OverwriteExisting or IgnoreIfExists.", mode => OverwriteMode = (OverwriteMode)Enum.Parse(typeof(OverwriteMode), mode, true));
-            options.Add("replace-existing", "If the package already exists in the repository, the default behavior is to reject the new package being pushed. You can pass this flag to overwrite the existing package. This flag may be deprecated in a future version; passing it is the same as using the OverwriteExisting overwrite-mode.", 
+            options.Add<OverwriteMode>("overwrite-mode=", "If the package already exists in the repository, the default behavior is to reject the new package being pushed (FailIfExists). You can use the overwrite mode to OverwriteExisting or IgnoreIfExists.", mode => OverwriteMode = mode);
```

Comment:
Should the list of valid values be generated from the enum now? E.g. https://github.com/OctopusDeploy/OctopusCLI/blob/7cbb503ac8720481735bf62b5b004719a5722178/source/Octopus.Cli/Commands/InstallAutoCompleteCommand.cs#L25